packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -195,6 +197,20 @@ export class ValidatorApi implements IValidatorApi {
     isAggregator: boolean
   ): Promise<void> {
     await checkSyncStatus(this.config, this.sync);
+    const currentSlot = this.chain.clock.currentSlot;
+    const epoch = computeEpochAtSlot(this.config, slot);
+    const currentEpoch = this.chain.clock.currentEpoch;
+    if (slot <= currentSlot) {
+      throw new ApiError(400, `Cannot prepare subnet for invalid slot ${slot}, current slot: ${currentSlot}`);
+    }
+    if (epoch > currentEpoch + 1) {
+      throw new ApiError(400, `Cannot prepare subnet for future epoch ${epoch}, current epoch: ${currentEpoch}`);
+    }
+    const headEpochCtx = await this.chain.getHeadEpochContext();
+    if (headEpochCtx.getCommitteeCountAtSlot(slot) !== committeesAtSlot) {
```

Comment:
for other clients, the point to have `committeesAtSlot` param is to avoid the node calculating it again (using `processSlots` and maybe `processEpoch`) but for us it's a simple get from EpochContext so I want an extra check here to potentially avoid DOS attack